File: src/main/java/com/back/nbe9112team06/domain/timetable/entity/TimeTable.java
package com.back.nbe9112team06.domain.timetable.entity;

import com.back.nbe9112team06.domain.meeting.entity.Meeting;
import com.back.nbe9112team06.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
public class TimeTable extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    private Meeting meeting;

    @OneToMany(mappedBy = "timeTable", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DateInfo> dateInfos = new ArrayList<>();


    public TimeTable(Meeting meeting, List<DateInfo> dateInfos){
        this.meeting = meeting;
        this.dateInfos = dateInfos;
    }
}

File: src/main/java/com/back/nbe9112team06/domain/timetable/entity/TimeTable.kt
package com.back.nbe9112team06.domain.timetable.entity

import com.back.nbe9112team06.domain.meeting.entity.Meeting
import com.back.nbe9112team06.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
class TimeTable() : BaseEntity() {

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id", unique = true)
    lateinit var meeting: Meeting

    @OneToMany(
        mappedBy = "timeTable",
        cascade = [CascadeType.ALL],
        orphanRemoval = true
    )
     var dateInfos: MutableList<DateInfo> = mutableListOf()

    constructor(
        meeting: Meeting,
        dateInfos: MutableList<DateInfo>
    ) : this() {
        this.meeting = meeting
        this.dateInfos = dateInfos
    }


}